app/controllers/dashboard.js

```diff
@@ -0,0 +1,59 @@
+const Departments = require('../models/departments');
+const Users = require('../models/users');
+const Messages = require('../models/messages');
+
+module.exports = {
+
+  orgIndex(req, res) {
+    let departments;
+    let departmentFilter = req.user.department || Number(req.query.department) || null;
+    let userFilter = req.query.user || null;
+    let users;
+    let countsByDay, countsByWeek;
+
+    Departments.findByOrg(req.user.org, true)
+    .then((depts) => {
+      departments = depts;
+
+      if (departmentFilter) {
+        if (req.user.department) departments = departments.filter((d) => { return d.department_id === departmentFilter});
+        return Users.findByDepartment(departmentFilter, true)
+      } else {
+        return Users.findByOrg(req.user.org, true)
+      }
+    }).then((u) => {
+      users = u;
+
+      if (departmentFilter) {
+        users = users.filter((u) => { return u.department == departmentFilter});
+        return Messages.countsByDepartment(req.user.org, departmentFilter, "day")
```

Comment:
let's even use long variable names here, perfect example of where it's confusing that `u` momentarily means `users` and `user`. Even on that one-line anonymous function, the clarity of `user.department` is helpful.